File: day-01/lua/part1.lua
#!/usr/bin/env lua5.4

local function parseInput()
    local num = {}
    for str in io.lines(arg[1]) do
        num[#num+1] = tonumber(str)
    end
    return num
end

local num = parseInput()
local target = 2020
-- local target = 99920044

local function part1()
    for i=1, #num do
        for j=1, #num do
            if num[i] + num[j] == target then
                return {num[i], num[j], num[i] * num[j]}
            end
        end
    end
end

local result = part1()

print(result[1].." * "..result[2].." = "..result[3])

File: day-01/python/part1.py
#!/usr/bin/env python3

import sys

num = [int(i) for i in open(sys.argv[1], 'r').read().split("\n")]
target = 2020
# target = 99920044

def part1():
    for i in num:
        for j in num:
            if i + j == target:
                assert i*j == 970816
                return [i, j, i*j]

result = part1()
print(f"{result[0]} * {result[1]} = {result[2]}")